packages/@aws-cdk/toolkit-lib/lib/api/cloud-assembly/source-builder.ts

```diff
@@ -379,30 +388,32 @@ export abstract class CloudAssemblySourceBuilder {
    */
   public async fromAssemblyDirectory(directory: string, props: AssemblyDirectoryProps = {}): Promise<ICloudAssemblySource> {
     const services: ToolkitServices = await this.sourceBuilderServices();
-    const contextAssemblyProps: ContextAwareCloudAssemblyProps = {
-      services,
-      contextStore: new MemoryContext(), // @todo We shouldn't be using a `ContextAwareCloudAssemblySource` at all.
-      lookups: false,
-    };
 
-    return new ContextAwareCloudAssemblySource(
-      {
-        produce: async () => {
-          // @todo build
-          await services.ioHelper.notify(IO.CDK_ASSEMBLY_I0150.msg('--app points to a cloud assembly, so we bypass synth'));
-
-          const readLock = await new RWLock(directory).acquireRead();
-          try {
-            const asm = await assemblyFromDirectory(directory, services.ioHelper, props.loadAssemblyOptions);
-            return new ReadableCloudAssembly(asm, readLock, { deleteOnDispose: false });
-          } catch (e) {
-            await readLock.release();
-            throw e;
+    return {
+      async produce() {
+        await services.ioHelper.notify(IO.CDK_ASSEMBLY_I0150.msg('--app points to a cloud assembly, so we bypass synth'));
+        const readLock = await new RWLock(directory).acquireRead();
+        try {
+          const asm = await assemblyFromDirectory(directory, services.ioHelper, props.loadAssemblyOptions);
+          const assembly = new ReadableCloudAssembly(asm, readLock, { deleteOnDispose: false });
+          if (assembly.cloudAssembly.manifest.missing && assembly.cloudAssembly.manifest.missing.length > 0) {
```

Comment:
Let me know if you want me to extract this into a new `ICloudAssemblySource`. I don't know that this needs to be reusable so I just inlined it here.

Reply:
inline is fine. we can extract this later